src/equations/compressible_euler_1d.jl

```diff
@@ -472,6 +472,83 @@ function flux_hllc(u_ll, u_rr, orientation::Integer, equations::CompressibleEule
 end
 
 
+"""
+  flux_hlle(u_ll, u_rr, orientation, equations::CompressibleEulerEquations1D)
+
+Computes the HLLE (Harten-Lax-van Leer-Einfeldt) flux for the compressible Euler equations.
+Special estimates of the signal velocites and linearization of the Riemann problem developed
+by Einfeldt to ensure that the internal energy and density remain positive during the computation
+of the numerical flux.
+
+Original publication:
+- Bernd Einfeldt (1988)
+  On Godunov-type methods for gas dynamics.
+  [DOI: 10.1137/0725021](https://doi.org/10.1137/0725021)
+
+Compactly summarized:
+- Siddhartha Mishra, Ulrik Skre Fjordholm and Rémi Abgrall
+  Numerical methods for conservation laws and related equations.
+  [Link](https://metaphor.ethz.ch/x/2019/hs/401-4671-00L/literature/mishra_hyperbolic_pdes.pdf)
+"""
+function flux_hlle(u_ll, u_rr, orientation::Integer, equations::CompressibleEulerEquations1D)
+  # Calculate primitive variables, enthalpy and speed of sound
+  rho_ll, v_ll, p_ll = cons2prim(u_ll, equations)
+  rho_rr, v_rr, p_rr = cons2prim(u_rr, equations)
+
+  # `u_ll[3]` is total energy `rho_e_ll` on the left
+  H_ll = (u_ll[3] + p_ll) / rho_ll
+  c_ll = sqrt(equations.gamma * p_ll / rho_ll)
+
+  # `u_rr[3]` is total energy `rho_e_rr` on the right
+  H_rr = (u_rr[3] + p_rr) / rho_rr
+  c_rr = sqrt(equations.gamma * p_rr / rho_rr)
+
+  # Compute Roe averages
+  sqrt_rho_ll = sqrt(rho_ll)
+  sqrt_rho_rr = sqrt(rho_rr)
+  inv_sum_sqrt_rho = inv(sqrt_rho_ll + sqrt_rho_rr)
+
+  v_roe = (sqrt_rho_ll * v_ll + sqrt_rho_rr * v_rr) * inv_sum_sqrt_rho
+  v_roe_mag = v_roe^2
+
+  H_roe = (sqrt_rho_ll * H_ll + sqrt_rho_rr * H_rr) * inv_sum_sqrt_rho
+  c_roe = sqrt((equations.gamma - 1) * (H_roe - 0.5 * v_roe_mag))
+
+  # Compute convenience constant for positivity preservation, see
+  # https://doi.org/10.1016/0021-9991(91)90211-3
+  beta = sqrt(0.5 * (equations.gamma - 1) / equations.gamma)
+
+  # Estimate the edges of the Riemann fan (with positivity conservation)
+  SsL = min(v_roe - c_roe, v_ll - beta * c_ll, 0.0 )
+  SsR = max(v_roe + c_roe, v_rr + beta * c_rr, 0.0 )
```

Comment:
```suggestion
  SsL = min(v_roe - c_roe, v_ll - beta * c_ll, zero(v_roe))
  SsR = max(v_roe + c_roe, v_rr + beta * c_rr, zero(v_roe))
```
Otherwise, you may run into type instabilities, e.g., when using AD. Please adapt the other dimensions as well.